### PR TITLE
feat(notifier-buy): shows only available plans for selection

### PR DESCRIPTION
**Pull request by itofarina**
## Description
Compares the available plans in cache with the active plans in the notifier provider instance to check if the plan is available for new subscriptions.

Resolves [RMKT-828](https://rsklabs.atlassian.net/jira/software/projects/RMKT/boards/60?selectedIssue=RMKT-828)

### CODE REVIEW

File: src/api/rif-notifier-service/subscriptionPlans.ts [hunk 1]
```diff
@@ -12,13 +12,19 @@ export default class SubscriptionPlans
...
+    if (!isValidResponse) {
+      this.errorReporter({
+        id: 'service-fetch',
+        text: 'Wrong response from notifier provider',
+        error: new Error(`Wrong response from notifier provider ${JSON.stringify(content)}`),
+      })
+    }
 
-    return isValidResponse && anyActive
+    return content.filter(({ planStatus }) => planStatus === 'ACTIVE')
```
Comment (vjgene) [inline src/api/rif-notifier-service/subscriptionPlans.ts]: we can use 

export const address = '/getSubscriptionPlans?activePlansOnly=true' as const

then

return content

File: src/api/rif-notifier-service/subscriptionPlans.ts [hunk 2]
```diff
@@ -12,13 +12,19 @@ export default class SubscriptionPlans
 
   _fetch = (): Promise<SubscriptionPlanResponse> => this.service.find()
 
-  hasActivePlans = async (): Promise<boolean> => {
+  getActivePlans = async (): Promise<Array<SubscriptionPlanDTO>> => {
```
Comment (vjgene) [inline src/api/rif-notifier-service/subscriptionPlans.ts:15]: Considering we are checking the length of the plans to determine 'hasActivePlans' in both ProviderRegistrar and NotifierOffersPage, why can't we return a boolean as before?

**Review by vjgene — COMMENTED** [review]
Looks good. Just couple of suggestions

File: src/api/rif-notifier-service/subscriptionPlans.ts (hunk 1 above)
Comment (itofarina) [reply src/api/rif-notifier-service/subscriptionPlans.ts]: Looks much better, will adapt, thanks.

File: src/api/rif-notifier-service/subscriptionPlans.ts (hunk 2 above)
Comment (itofarina) [reply src/api/rif-notifier-service/subscriptionPlans.ts:15]: For now we are only using the length, but I just changed this with the idea of comparing the plans we are allowing to select in the checkout with this result.
If the plans are not in this result then they shouldn't be selectable. So we need them for the comparison.

**Review by vjgene — APPROVED** [review]
Looks good!

**Pull request merged**